src/backend/src/user_profile.rs

```diff
@@ -3,9 +3,24 @@ use ic_cdk::api::time;
 use ic_stable_structures::StableBTreeMap;
 use shared::types::{
     user_profile::{AddUserCredentialError, GetUserProfileError, StoredUserProfile},
-    CredentialType, Version,
+    CredentialType, Timestamp, Version,
 };
 
+fn store_user(
+    principal: StoredPrincipal,
+    timestamp: Timestamp,
+    user: &StoredUserProfile,
+    user_profile_map: &mut StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
```

Comment:
nth: if we create types for those StableBTreeMaps it would make the parameters a bit less verbose

Reply:
I agree and I can do it in another PR. It requires moving some types around because some are more private and can't be moved so easily.